extensions/amp-form/0.1/amp-form.js

```diff
@@ -276,6 +278,11 @@ export class AmpForm {
       for (let i = 0; i < varSubsFields.length; i++) {
         this.urlReplacement_.expandInputValueSync(varSubsFields[i]);
       }
+
+      // If this wasn't a submission event, trigger form submit.
```

Comment:
Yeah this confusion comes from reusing this method once for actual `submit` event handling and the other for doing an actual submission flow (triggered by `on=form.submit` for example).

I couldn't figure out how to refactor the function to reuse the functionality in either cases, but I can think a bit more about it and see if I can simplify it.